lib/compare_with_wikidata/diff_row.rb

```diff
@@ -64,8 +64,13 @@ def value_cells
         # Expand Wikidata IDs to templates.
         # This expands cases where the whole cell is just a Q value:
         value = v.to_s.sub('http://www.wikidata.org/entity/', '').sub(/^Q(\d+)$/, '{{Q|\\1}}')
-        # This expands cases where the whole cell is a change in Q value:
-        value.sub!(/^Q(\d+)->Q(\d+)$/, '{{Q|\\1}}->{{Q|\\2}}')
+        # This expands cases where the whole cell is a change in Q
+        # value (and either side of the change might be NULL)
+        if value =~ /^(Q\d+|NULL)->(Q\d+|NULL)$/
+          value = Regexp.last_match.captures.map do |c|
+            c == 'NULL' ? 'NULL' : "{{Q|#{c[1..-1]}}}"
+          end.join('->')
+        end
```

Comment:
I'm finding this approach a little bit difficult to follow. I wonder if it might be simpler to expand line 66 to something more like:

```ruby
value = v.to_s.sub('http://www.wikidata.org/entity/', '').split('->').map { |e| e.sub(/^Q(\d+)$/, '{{Q|\\1}}') }.join('->')
```

That's very slightly different behaviour, but I'm not sure it's undesirable, and the tests all pass with it.

Reply:
Funnily enough, I woke up in the night and thought "ugh, that change I made was ugly, I bet Tony wouldn't like it" :)

I like this suggestion - I've pushed a fixup based on it.